src/uu/snice/src/snice.rs

```diff
@@ -66,86 +69,25 @@ impl SignalDisplay {
     }
 }
 
-#[derive(Debug)]
-struct Settings {
-    display: Option<SignalDisplay>,
-    expressions: Option<Vec<SelectedTarget>>,
-    priority: Priority,
-    verbose: bool,
-}
-
-impl Settings {
-    fn try_new(matches: &ArgMatches) -> UResult<Self> {
-        let priority = matches
-            .try_get_one::<String>("priority")
-            .unwrap_or(Some(&String::new()))
-            .cloned();
-
-        let expression = match priority {
-            Some(expr) => {
-                Priority::try_from(expr).map_err(|err| USimpleError::new(1, err.to_string()))?
-            }
-            None => Priority::default(),
-        };
-
-        let display = if matches.get_flag("table") {
-            Some(SignalDisplay::Table)
-        } else if matches.get_flag("list") {
-            Some(SignalDisplay::List)
-        } else {
-            None
-        };
+pub fn print_signals(settings: &Settings) -> Option<()> {
```

Comment:
The return type is unexpected. I think it is not necessary if you make the function do what its name says: to print signals. And to move the decision, whether the user wants to print the signals, to the caller of the function. Something like:
```rust
// in uumain
if let Some(display) = settings.display {
    print_signals(display);
    return Ok(());
}
```